tests.js

```diff
@@ -96,17 +127,82 @@ var main = function(reducer){
     });
 };
 
-
 // This is the function to perform the minification in
 // It takes an object, and should return an object
 // the goal is to preserve the meaning on the object
-var reducer = function(object){
-    return object;
+var reducer = function(data){
+    var object = Object.assign({}, data);
+    if (object['ctor']) {
+      if ((object['ctor']).charAt(0) === '_') {
+        // tuple
+        delete object['ctor'];
+        var data = [];
+        Object.keys(object).forEach(function(key) {
```

Comment:
So I actually think instead of doing `delete object['ctor']`, it might be nicer to do:

```
Objects.keys(object).forEach(function(key){
  if (key === 'ctor') { return; }

  var i = parseInt(key.slice(1));
  data[i] = object[key];
});
```
since that way there's less mutation going on